### PR TITLE
API: Add sparse Acessor

### LINKED ISSUE #23148: ENH: Add sparse Accessor for SparseDtype data
This is a blocker for removing SparseDataFrame and SparseSeries.

We'd register it for Series and DataFrame I think.

```python

In [4]: [x for x in set(dir(pd.SparseSeries)) - set(dir(pd.Series)) if not x.startswith('_')]
Out[4]:
['from_coo',
 'block',
 'npoints',
 'sparse_reindex',
 'kind',
 'density',
 'as_sparse_array',
 'fill_value',
 'to_coo',
 'sp_index',
 'sp_values']

```

```python
In [15]: [x for x in set(dir(pd.DataFrame)) - set(dir(pd.DataFrame)) if not x.startswith('_')]
Out[15]: []

```

### PR DESCRIPTION
**Pull request by TomAugspurger**
* Adds a Series.sparse accessor
* Adds several methods to SparseArray to for use via the accessor

Closes #23148.

This should provide all the methods / attributes that were available on SparseSeries, but not Series.

Right now the docs for `.sparse.from_coo` and `to_coo` seem to be broken. It's a bit strange since they're implemented on the accessor (they don't make sense on the Array)

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Latest commit should fix the doc rendering for Series.sparse.to_coo and Series.sparse.from_coo (we just need to use the regular `autosummary` directive, instead of the `autoaccessor`, since the docstring is on the accessor, not the object delegated to).

File: doc/source/sparse.rst [hunk 1]
```diff
@@ -62,6 +62,24 @@ Any sparse object can be converted back to the standard dense form by calling
 
    sts.to_dense()
 
+.. _sparse.accessor:
```
Comment (jreback) [inline doc/source/sparse.rst]: versionadded here

File: doc/source/whatsnew/v0.24.0.txt [hunk 2]
```diff
@@ -427,14 +427,20 @@ changes were made:
...
 
-
 Some new warnings are issued for operations that require or are likely to materialize a large dense array:
 
 - A :class:`errors.PerformanceWarning` is issued when using fillna with a ``method``, as a dense array is constructed to create the filled array. Filling with a ``value`` is the efficient way to fill a sparse array.
 - A :class:`errors.PerformanceWarning` is now issued when concatenating sparse Series with differing fill values. The fill value from the first sparse array continues to be used.
 
 In addition to these API breaking changes, many :ref:`performance improvements and bug fixes have been made <whatsnew_0240.bug_fixes.sparse>`.
 
+Finally, a ``Series.sparse`` accessor has added to provide sparse-specific methods and attributes.
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: can you add link to docs

File: doc/source/api.rst [hunk 3]
```diff
@@ -851,6 +851,24 @@ Sparse
    SparseSeries.to_coo
    SparseSeries.from_coo
 
+.. autosummary::
+   :toctree: generated/
+   :template: autosummary/accessor_attribute.rst
+
+   Series.sparse.npoints
+   Series.sparse.kind
```
Comment (jreback) [inline doc/source/api.rst]: nbytes?

**Review by jreback — COMMENTED** [review]
lgtm. some minor comments.

File: doc/source/whatsnew/v0.24.0.txt (hunk 2 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v0.24.0.txt]: Can't link to a generic `Series.sparse` location, but  I can do

    to provide sparse-specific methods like :meth:`Series.sparse.from_coo`.

and people can navigate from there.

File: doc/source/api.rst (hunk 3 above)
Comment (TomAugspurger) [reply doc/source/api.rst]: That should already be reflected in Series.nbytes.

File: pandas/core/sparse/series.py [hunk 4]
```diff
@@ -581,98 +580,13 @@ def combine_first(self, other):
         return dense_combined.to_sparse(fill_value=self.fill_value)
 
     def to_coo(self, row_levels=(0, ), column_levels=(1, ), sort_labels=False):
-        """
-        Create a scipy.sparse.coo_matrix from a SparseSeries with MultiIndex.
```
Comment (jorisvandenbossche) [inline pandas/core/sparse/series.py]: Can you keep those docstrings here (or share it with the delegate method)

File: doc/source/whatsnew/v0.24.0.txt [hunk 5]
```diff
@@ -521,14 +521,20 @@ changes were made:
...
 
-
 Some new warnings are issued for operations that require or are likely to materialize a large dense array:
 
 - A :class:`errors.PerformanceWarning` is issued when using fillna with a ``method``, as a dense array is constructed to create the filled array. Filling with a ``value`` is the efficient way to fill a sparse array.
 - A :class:`errors.PerformanceWarning` is now issued when concatenating sparse Series with differing fill values. The fill value from the first sparse array continues to be used.
 
 In addition to these API breaking changes, many :ref:`performance improvements and bug fixes have been made <whatsnew_0240.bug_fixes.sparse>`.
 
+Finally, a ``Series.sparse`` accessor has added to provide sparse-specific methods like :meth:`Series.sparse.from_coo`.
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.24.0.txt]:
```suggestion
Finally, a ``Series.sparse`` accessor was added to provide sparse-specific methods like :meth:`Series.sparse.from_coo`.
```

**Review by jorisvandenbossche — COMMENTED** [review]
I am fine with the idea of a sparse accessor. I only have some hesitation about some of the attributes exposed there, as this enforces them as public API. 
(as I think I already mentioned before, promoting this from SparseSeries to Series is like more officially supporting this as a stable part of pandas).

For example, do we want to expose the IntIndex and BlockIndex officially as API?

**Comment by TomAugspurger** [comment]
> For example, do we want to expose the IntIndex and BlockIndex officially as API?

I'm happy to remove those.

File: pandas/core/sparse/series.py (hunk 4 above)
Comment (TomAugspurger) [reply pandas/core/sparse/series.py]: My hope is to deprecate SparseSeries entirely for 0.24.0. But I suppose we'll want a docstring for the deprecation period, so I'll use shared docs.

**Comment by TomAugspurger** [comment]
Forgot to remove sp_index and kind from the api.rst

This should be ready once it passes.

**Comment by jreback** [comment]
thanks @TomAugspurger

**Pull request merged**